src/theme.tsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * This code is extracted from here https://www.joshwcomeau.com/react/dark-mode/
+ * I deleted the parts I'm not using though, since I just want the user theme.
+ */
+import React from "react";
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const ThemeContext = React.createContext<
+  | {
+      colorMode: string;
+    }
+  | undefined
+>(undefined);
+
+function getInitialColorMode() {
+  const mql = window.matchMedia("(prefers-color-scheme: dark)");
+  const hasMediaQueryPreference = typeof mql.matches === "boolean";
+  if (hasMediaQueryPreference) {
```

Comment:
praise: Well done with the theming setup here!

bug: It's just missing the mq listener, in case the user changes the theme while the page is running.